grammar.js

```diff
@@ -70,736 +76,751 @@ module.exports = grammar({
+
+    loop_label_inverse: ($) =>
+      seq(":", field("name", alias($.identifier, $.label_identifier))),
+
+    return_expression: ($) =>
+      choice(prec.left(seq("return", $._expression)), prec(-1, "return")),
+
+    test_expression: ($) =>
+      seq("test", field("test_name", $.string_literal), field("body", $.block)),
+
+    try_expression: ($) =>
+      prec.left(seq("try", field("expression", $._expression))),
+
+    usingnamespace_expression: ($) =>
+      prec.left(
+        seq(
+          optional($.visibility_modifier),
+          "usingnamespace",
+          field("import", $._expression)
+        )
+      ),
+
+    _type: ($) =>
+      prec(
+        -1,
+        choice(
+          $.primitive_type,
+          $.optional_type,
+          $.pointer_type,
+          $.error_type,
+          $.array_type,
+          $.custom_number_type,
+          alias($.identifier, $.type_identifier)
+        )
+      ),
+
+    primitive_type: (_) => choice(...primitive_types),
+
+    custom_number_type: (_) => /(i|u)[0-9]+/,
 
     // Having the err field optional cause a bunch of conflicts
     // so when `!<type>` happens, it will show as unary expression
-    error_type: $ => prec.left(seq(
-      field('err', $._type),
-      '!',
-      field('ok', $._type),
-    )),
-
-    optional_type: $ => prec(PREC.optional, seq(
-      '?',
-      $._type,
-    )),
-
-    array_type: $ => prec(PREC.array, seq(
-      repeat(seq(
-        '[',
-        field('size', optional(choice(
-          $.integer_literal,
-          $.identifier,
-          alias(seq('*', optional('c')), $.pointer)))),
-        ']',
-      )),
-      optional(repeat($.type_prefix)),
-      // Cannot be another array type, since the sintax already cover muldimentional arrays
-      choice(
-        $.primitive_type,
-        $.optional_type,
-        $.pointer_type,
-        $.error_type,
-        $.custom_number_type,
+    error_type: ($) =>
+      prec.left(seq(field("err", $._type), "!", field("ok", $._type))),
+
+    optional_type: ($) => prec(PREC.optional, seq("?", $._type)),
+
+    array_type: ($) =>
+      prec(
+        PREC.array,
+        seq(
+          repeat(
+            seq(
+              "[",
+              field(
+                "size",
+                optional(
+                  choice(
+                    $.integer_literal,
+                    $.identifier,
+                    alias(seq("*", optional("c")), $.pointer)
+                  )
+                )
+              ),
+              "]"
+            )
+          ),
+          optional(repeat($.type_prefix)),
+          // Cannot be another array type, since the sintax already cover
+          // muldimentional arrays
+          choice(
+            $.primitive_type,
+            $.optional_type,
+            $.pointer_type,
+            $.error_type,
+            $.custom_number_type,
+            alias($.identifier, $.type_identifier)
+          )
+        )
+      ),
```

Comment:
The formatting here was kinda too much, there is a way to add a config that doesn't do that kinda of thing (Sorry, I have no idea, never used a formater for JS before)?  There some other parts that became like that, but that was the first that caught my attention

Reply:
Woah. didn't see that. That's way too nested. I just ran `prettier grammar.js`. There are a few configs available for prettier albeit i would still suggest going for the default prettier config. I know that part looks horrible. But well. Let's keep it that way, what do you say? Since that's the default.